### PR TITLE
Fix build on Travis

### PR DESCRIPTION
**Pull request by mvz**
## Summary

This is a follow-up of #433. It should fix all the failing scenarios that are now properly detected by Travis.

## Details

This change:
* Ensures a failed build is detected as such (at lease as far as the cucumber features are concerned).
* Fixes the build failures that were due to renaming of `cli` to `aruba-test-cli` in all the features (
* Fixes the `cucumber/setup.feature` feature that was broken from the start because it didn't require the needed part of Aruba.
* Fixes the configuration features by requiring the necessary part of Aruba. This was broken because the features replaced the default file.

## Motivation and Context

Builds were passing even though the cucumbers were failing. This made the Travis CI results effectively useless. Rather than merging in just the PR that makes the build break (#433) , this makes it pass so there is a solid base for future development.

### CODE REVIEW

**Comment by mvz** [comment]
Additionally, this adds the 'debug' group to `licence_finder`'s ignore list, since that group is not installed on Travis CI.

**Comment by mvz** [comment]
Left to do:
* [ ] Fix failing scenario on JRuby
* [ ] Speed up cukes so they stay under the 50m time limit.

**Comment by mvz** [comment]
The failing scenario seems to have something to do with the combination of childprocess en non-ascii text.

**Comment by mvz** [comment]
In fact, the JRuby failure looks like it's the same issue as #417.

**Comment by maxmeyer** [comment]
@mvz Thanks a lot. Would you mind to rebase? I oversaw your PR and fix the `aruba-test-cli`-isse myself.

File: lib/aruba/processes/spawn_process.rb [hunk 1]
```diff
@@ -267,7 +270,7 @@ def read_temporary_output_file(file)
         data = file.read
         file.close
 
-        data
+        data.force_encoding('UTF-8')
```
Comment (maxmeyer) [inline lib/aruba/processes/spawn_process.rb]: @enkessler There seem to be some encoding issues with childprocess. Can you help us with that?

**Comment by mvz** [comment]
@maxmeyer No problem, I rebased it.

File: lib/aruba/processes/spawn_process.rb (hunk 1 above)
Comment (enkessler) [inline lib/aruba/processes/spawn_process.rb]: I can try.

**Comment by mvz** [comment]
Some new build issue has appeared to to the rebase. I'll take a look.

**Comment by mvz** [comment]
@maxmeyer I'm a little confused: Are the deprecated functions supposed to be available in the upcoming version 1.0.0 or not?

**Comment by maxmeyer** [comment]
> Are the deprecated functions supposed to be available in the upcoming version 1.0.0 or not?

@mvz Yes, they're all done when we release 1.0.0. But this should be done in a separate PR

File: features/02_configure_aruba/activate_announcer_on_command_failure.feature [hunk 2]
```diff
@@ -10,6 +10,8 @@ Feature: Configure announcer activation on command failure
   Scenario: Default value
     Given a file named "features/support/aruba.rb" with:
```
Comment (maxmeyer) [inline features/02_configure_aruba/activate_announcer_on_command_failure.feature]: Let's rename the file to say `aruba_config.rb` instead of adding `require ...`. This is not the thing we want to show the user in this example. The require is done in the fixture in `aruba.rb`.

File: features/02_configure_aruba/activate_announcer_on_command_failure.feature [hunk 3]
```diff
@@ -23,6 +25,8 @@ Feature: Configure announcer activation on command failure
   Scenario: Modify value
     Given a file named "features/support/aruba.rb" with:
     """ruby
+    require 'aruba/cucumber'
```
Comment (maxmeyer) [inline features/02_configure_aruba/activate_announcer_on_command_failure.feature]: Same here.

File: features/02_configure_aruba/command_runtime_environment.feature [hunk 4]
```diff
@@ -19,6 +19,7 @@ Feature: Define default process environment
     Given a file named "spec/environment_spec.rb" with:
     """ruby
     require 'spec_helper'
+    require 'aruba/rspec'
```
Comment (maxmeyer) [inline features/02_configure_aruba/command_runtime_environment.feature]: Why did you add this? Do you see an error? That's should already provided by the fixture in `spec/support/aruba.rb`.

File: features/02_configure_aruba/command_runtime_environment.feature [hunk 5]
```diff
@@ -40,6 +41,7 @@ Feature: Define default process environment
...
```
Comment (maxmeyer) [inline features/02_configure_aruba/command_runtime_environment.feature]: Same here

File: features/02_configure_aruba/command_runtime_environment.feature [hunk 6]
```diff
@@ -63,6 +65,7 @@ Feature: Define default process environment
...
```
Comment (maxmeyer) [inline features/02_configure_aruba/command_runtime_environment.feature]: Same here

File: features/02_configure_aruba/command_runtime_environment.feature [hunk 7]
```diff
@@ -86,6 +89,7 @@ Feature: Define default process environment
...
```
Comment (maxmeyer) [inline features/02_configure_aruba/command_runtime_environment.feature]: Same here.

File: features/02_configure_aruba/command_runtime_environment.feature [hunk 8]
```diff
@@ -109,6 +113,7 @@ Feature: Define default process environment
...
```
Comment (maxmeyer) [inline features/02_configure_aruba/command_runtime_environment.feature]: Same here.

File: lib/aruba/api.rb [hunk 9]
```diff
@@ -8,7 +8,7 @@
 require 'aruba/api/core'
 require 'aruba/api/command'
 
-if Aruba::VERSION <= '1.0.0'
+if Aruba::VERSION <= '1.1.0'
```
Comment (maxmeyer) [inline lib/aruba/api.rb]: I think you add this to make this PR not to fail the build, right?

File: lib/aruba/processes/spawn_process.rb [hunk 10]
```diff
@@ -64,6 +64,9 @@ def start
         @stdout_file.sync = true
         @stderr_file.sync = true
 
+        @stdout_file.binmode
+        @stderr_file.binmode
```
Comment (maxmeyer) [inline lib/aruba/processes/spawn_process.rb]: What happen's if this is not binmode?

File: spec/event_bus_spec.rb [hunk 11]
```diff
@@ -1,3 +1,4 @@
+require 'spec_helper'
 require 'aruba/event_bus'
```
Comment (maxmeyer) [inline spec/event_bus_spec.rb]: Is this line really required after your change?

File: spec/aruba/hooks_spec.rb [hunk 12]
```diff
@@ -1,3 +1,4 @@
+require 'spec_helper'
 require 'aruba/config'
```
Comment (maxmeyer) [inline spec/aruba/hooks_spec.rb]: Can we remove this line after your change?

File: spec/aruba/processes/basic_process_spec.rb [hunk 13]
```diff
@@ -1,3 +1,4 @@
+require 'spec_helper'
 require 'aruba/processes/basic_process'
```
Comment (maxmeyer) [inline spec/aruba/processes/basic_process_spec.rb]: Same here.

File: spec/aruba/spawn_process_spec.rb [hunk 14]
```diff
@@ -1,3 +1,4 @@
+require 'spec_helper'
 require 'aruba/processes/spawn_process'
```
Comment (maxmeyer) [inline spec/aruba/spawn_process_spec.rb]: Same here.

File: spec/event_bus/name_resolver_spec.rb [hunk 15]
```diff
@@ -1,3 +1,4 @@
+require 'spec_helper'
 require 'aruba/event_bus/name_resolver'
```
Comment (maxmeyer) [inline spec/event_bus/name_resolver_spec.rb]: Same here.

File: lib/aruba/processes/spawn_process.rb (hunk 10 above)
Comment (maxmeyer) [inline lib/aruba/processes/spawn_process.rb]: Ah ok. Thanks for the helpful commit message.

**Comment by mvz** [comment]
> @mvz Yes, they're all done when we release 1.0.0. But this should be done in a separate PR

Ok, for that purpose I have updated the version checks to allow deprecated functions when the version is 1.0.0 for now, since otherwise everything breaks at once. The removal can then safely be finalized in that separate PR.

File: features/02_configure_aruba/activate_announcer_on_command_failure.feature (hunk 2 above)
Comment (mvz) [inline features/02_configure_aruba/activate_announcer_on_command_failure.feature]: Yes, that makes sense.

**Comment by mvz** [comment]
Since the version updated causes some new spec failures, I propose the following:
* I create a new PR with just the parts that make the specs run (and fail in one or two places).
* Next, I'm going to fix the one or two failing specs (I'm partway done with this) and make a second PR for that.
* After that, I can update this one and see which bits are still needed.

File: lib/aruba/api.rb (hunk 9 above)
Comment (mvz) [inline lib/aruba/api.rb]: Yes, Aruba still uses and/or tests some deprecated functions at the moment.

File: spec/aruba/hooks_spec.rb (hunk 12 above)
Comment (mvz) [inline spec/aruba/hooks_spec.rb]: No, it seems it can go.

**Comment by maxmeyer** [comment]
Thanks @mvz. BTW: I like your commit style. 

Would you mind to squash those commits handling the same lines of code - e.g. introduce the requires + get rid of those requires. Alternatively drop them.

I will have a detailed look at your PR tomorrow.

File: lib/aruba/api.rb (hunk 9 above)
Comment (maxmeyer) [inline lib/aruba/api.rb]: Yes, might be. Some months ago I started getting rid of them, but somehow I managed this branch to get lost. Since I move my "working environment". But I'm quite eager to remove them as fast as possible. There are tons of (very) old code which is deprecated for ages.

**Comment by maxmeyer** [comment]
> In fact, the JRuby failure looks like it's the same issue as #417.

Would be great if this is fixed with your PR. JRuby have given me a few headaches so far. I'm just on C-Ruby. :smile:

File: features/02_configure_aruba/command_runtime_environment.feature (hunk 4 above)
Comment (mvz) [inline features/02_configure_aruba/command_runtime_environment.feature]: No, it seems this not needed. I'm not sure what made me add it.

**Comment by mvz** [comment]
@maxmeyer do you still want to have the tmp dir used depend on the suite (rspec vs cucumber)? It seems to me to depend too much on the end user's chosen set up whether that's useful (plus it breaks some of the current scenarios, which is why I'm asking).

File: lib/aruba/cucumber/hooks.rb [hunk 16]
```diff
@@ -5,7 +5,7 @@
 
 if Aruba::VERSION >= '1.0.0'
   Around do |_, block|
-    Aruba.platform.with_environment(&block)
+    with_environment(&block)
```
Comment (maxmeyer) [inline lib/aruba/cucumber/hooks.rb]: I did this change before and something was uterly broken afterwards - I was forced to revert that change. What's the reason for this change?

File: lib/aruba/cucumber/core.rb [hunk 17]
```diff
@@ -1,4 +1,4 @@
-if Aruba::VERSION >= '1.0.0'
+if false && Aruba::VERSION >= '1.0.0'
```
Comment (maxmeyer) [inline lib/aruba/cucumber/core.rb]: Personally I use both `rspec` and `cucumber` in projects. Sometimes I ran both and want the results to be availabe. That's the reason for this fix. Can you point me to the failure to see if there's something we can do about this?

File: features/05_use_rspec_matchers/command/run_too_long.feature [hunk 18]
```diff
@@ -25,3 +25,24 @@ Feature: Check if a timeout occured during command execution
     """
     When I run `rspec`
     Then the specs should all pass
+
```
Comment (maxmeyer) [inline features/05_use_rspec_matchers/command/run_too_long.feature]: This should be on its own feature file. :astonished:.

https://github.com/cucumber/aruba/blob/master/features/05_use_rspec_matchers/command/have_finished_within_time.feature

Did you rebase your PR?

File: spec/aruba/api_spec.rb [hunk 19]
```diff
@@ -102,6 +102,26 @@
         expect(File.exist?(File.expand_path(@directory_path))).to be_truthy
       end
     end
+
+    describe '#cd' do
```
Comment (maxmeyer) [inline spec/aruba/api_spec.rb]: Let's move the changes for env to a separate PR.

File: lib/aruba/rspec.rb [hunk 20]
```diff
@@ -18,6 +18,12 @@
     if self.class.include? Aruba::Api
       restore_env
       setup_aruba
+
```
Comment (maxmeyer) [inline lib/aruba/rspec.rb]: This whole changing environment-thing in aruba is a bit of a hassle. I would love to this in a different PR, please.

File: spec/aruba/api_spec.rb [hunk 21]
```diff
@@ -1071,6 +1071,48 @@ def actual_permissions
 
         expect(ENV[variable]).to eq '1'
       end
+
```
Comment (maxmeyer) [inline spec/aruba/api_spec.rb]: I'm interested to see, if this #405 is fixed by your change. Would be quite happy if the answer is yes. Yet again, please move this to another PR.

**Comment by maxmeyer** [comment]
@mvz I would like to merge this PR before we/you go on fixing more bugs. Can you please extract the commits handling env logic to a separate PR. I'm going to merge this one then.

File: spec/aruba/api_spec.rb (hunk 21 above)
Comment (maxmeyer) [inline spec/aruba/api_spec.rb]: Forget about my comment extracting the env thing. I just saw I already created a PR for that.

File: lib/aruba/cucumber/hooks.rb (hunk 16 above)
Comment (mvz) [inline lib/aruba/cucumber/hooks.rb]: The `Aruba.platform.with_environment` version clears the entire environment, including things like `PATH`, causing a lot of cukes to fail because they can't find executables, among other things.

Comment (mvz) [inline lib/aruba/cucumber/hooks.rb]: I'll move this commit to the PR about env handling, by the way.

File: features/05_use_rspec_matchers/command/run_too_long.feature (hunk 18 above)
Comment (mvz) [inline features/05_use_rspec_matchers/command/run_too_long.feature]: Ah, that's where it went 😄. I'll remove this duplicate.

File: lib/aruba/cucumber/core.rb (hunk 17 above)
Comment (mvz) [inline lib/aruba/cucumber/core.rb]: There are several cucumber scenario's that check directories. These fail because they expect `tmp/aruba`, and not `tmp/cucumber`. This is fairly easy to change, but to me it makes more sense to let each user of aruba to set their own working directory if necessary, since there are many ways to parallelize the build, e.g., by running several instances of cucumber at the same time.

**Comment by mvz** [comment]
I rebased this and isolated all env-related commits in #442. I'll hold of on adding more fixes for now so we can get some of this merged. I'm happy to create more separate PRs since this is quite a mixed bag of commits.

File: lib/aruba/cucumber/core.rb (hunk 17 above)
Comment (maxmeyer) [inline lib/aruba/cucumber/core.rb]: Ok, then let's remove the lines altogether instead of using `false`.

File: lib/aruba/cucumber/hooks.rb (hunk 16 above)
Comment (maxmeyer) [inline lib/aruba/cucumber/hooks.rb]: ok

**Comment by maxmeyer** [comment]
@mvz Personally I would love to see all env related commits to be moved to that separate PR - if not already done - and then dropped from this PR. I think there a few left in here.

After the minor other things I just saw today, we're good to merge this one. After that I will have a look into the env PR.

**Comment by mvz** [comment]
> and then dropped from this PR. I think there a few left in here.

Yes, all the commits of #442 are also in this PR. It's probably best to first merge #442 and then rebase this one, but I can separate them out if you prefer.

File: lib/aruba/cucumber/core.rb (hunk 17 above)
Comment (mvz) [inline lib/aruba/cucumber/core.rb]: Ok, I'll do that.

**Comment by mvz** [comment]
I have now rebased this so it no longer includes the environment fixes that are in #442.

**Comment by maxmeyer** [comment]
@mvz I'm going to merge this PR first. But please do not wonder, if this happens just at the beginning of next week.

BTW: Did you see my comments about the `aruba/cucumber` vs. `aruba/api`-thing? I just pressed the review finished button.

**Comment by mvz** [comment]
> Did you see my comments about the aruba/cucumber vs. aruba/api-thing?

Ah yes, they only showed up just now. I'll take a look.

File: features/03_testing_frameworks/cucumber/setup_aruba_for_cucumber.feature [hunk 22]
```diff
@@ -47,7 +47,7 @@ Feature: Getting started with Cucumber and aruba
 
     Given a file named "features/support/env.rb" with:
     """
-    require 'aruba/api'
+    require 'aruba/cucumber'
```
Comment (mvz) [inline features/03_testing_frameworks/cucumber/setup_aruba_for_cucumber.feature]: Ok, clear. I have removed that change.

**Comment by mvz** [comment]
🎉
